mmv1/third_party/terraform/utils/kms_utils.go

```diff
@@ -188,16 +220,37 @@ func clearCryptoKeyVersions(cryptoKeyId *kmsCryptoKeyId, userAgent string, confi
 	}
 
 	for _, version := range versionsResponse.CryptoKeyVersions {
-		request := &cloudkms.DestroyCryptoKeyVersionRequest{}
-		destroyCall := versionsClient.Destroy(version.Name, request)
-		if config.UserProjectOverride {
-			destroyCall.Header().Set("X-Goog-User-Project", cryptoKeyId.KeyRingId.Project)
+		// skip the versions that have been destroyed earlier
+		if version.State == "ENABLED" {
```

Comment:
It seems like there's a small chance of a race condition here - i.e. if the state is `PENDING_GENERATION`, can it be destroyed, or can it only be destroyed if `ENABLED`?

This would also skip destroying versions that are `DISABLED` - I am pretty sure we should continue destroying those.

Perhaps it would be better to explicitly check if the state is `DESTROYED` or `DESTROY_SCHEDULED` if those are what are causing errors - what do you think?

Reply:
For this specific change, I'm adding it to not create a merge conflict against the current master branch. Sam pushed this previously. 
I agree with you that we should be more specific on checking the different state for the cryptoKeyVersions.